Analytical_Engine/Query/IsAligned.cs

```diff
@@ -61,7 +61,7 @@ public static partial class Query
             Vector planeNormal = plane.Normal;
             Vector polycurveNormal = polycurve.Normal();
 
-            return planeNormal.Normalise().IsParallel(polycurveNormal.Normalise(), tolerance) == 1;
+            return planeNormal.IsParallel(polycurveNormal, tolerance) == 1 || planeNormal.IsParallel(polycurveNormal, tolerance) == -1;
```

Comment:
cant we just check != 0 instead :). Avoid checking IsParalell twice.

```suggestion
            return planeNormal.IsParallel(polycurveNormal, tolerance) != 0;
```